src/js/components/AppPageComponent.jsx

```diff
@@ -343,11 +343,8 @@ var AppPageComponent = React.createClass({
     var volumes = AppsStore.getVolumes(model.id);
 
     var tabs = state.tabs.filter(tab =>
-      tab.text !== "Volumes" || (
-        model.container != null &&
-        model.container.volumes != null &&
-        model.container.volumes.length > 0
-      ));
```

Comment:
@aldipower just to clarify: the new `if` condition will be false if the app has no `tasks` at all. It will be true if there is at  least one task and one volume, no matter if the task is running or not. 
I think the new `if` is right, as I don't see how we could have a volume without any task.